### PR TITLE
Fix Blockly variable prompt dialog bug

### PR DESCRIPTION
**Pull request by microsoftsam**
Issue with the Blockly prompt dialog when the defaultValue is set. 
We always return the defaultValue whereas we should always return "" if cancel is pressed. 
If okay pressed, we should return the value in the field, and if defaultValue is set, we should just set the initial value to be the default value (instead of what we do now which is set it as the placeholder). 

this is all in sync with what a window.prompt would do.

### CODE REVIEW

File: webapp/src/core.ts [hunk 1]
```diff
@@ -341,13 +341,14 @@ export function promptAsync(options: PromptOptions): Promise<string> {
...
     options.htmlBody = `<div class="ui fluid icon input">
-                            <input type="text" id="promptDialogInput" placeholder="${options.defaultValue}">
+                            <input type="text" id="promptDialogInput" value="${options.defaultValue}">
                         </div>`;
 
     options.onLoaded = () => {
         let dialogInput = document.getElementById('promptDialogInput') as HTMLInputElement;
         if (dialogInput) {
             dialogInput.focus();
+            dialogInput.selectionStart = dialogInput.selectionEnd = 10000;
```
Comment (pelikhan) [inline webapp/src/core.ts]: ?

Comment (microsoftsam) [inline webapp/src/core.ts]: Hack to set the selection to the end, apparently works on all browsers.

**Comment by pelikhan** [comment]
Isn't it better to select the entire text? Makes it easier to rename.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by microsoftsam** [comment]
Yeah probably, let me take a look.

**Pull request merged**